Polyadic.Json/NewType/JsonNewTypeJsonConverterFactory.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Polyadic.Json.NewType
+{
+    internal sealed class JsonNewTypeJsonConverterFactory : JsonConverterFactory
+    {
+        public override bool CanConvert(Type typeToConvert) => Attribute.IsDefined(typeToConvert, typeof(JsonNewTypeAttribute));
+
+        public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+        {
+            var metadata = GetMetadata(typeToConvert, options);
+            return (JsonConverter)Activator.CreateInstance(
+                typeof(JsonNewTypeConverter<,>).MakeGenericType(typeToConvert, metadata.InnerType),
+                metadata)!;
+        }
+
+        private static NewTypeMetadata GetMetadata(Type typeToConvert, JsonSerializerOptions options)
+        {
+            var constructor = FindConstructor(typeToConvert);
+            var innerType = constructor.GetParameters()[0].ParameterType;
```

Comment:
Is there a reason you use the array operator instead of First() ?

Reply:
44edcd7e993f8c034a31e140429837c0b1d65b30